### PR TITLE
[CosmosDB] `az cosmosdb sql container update`: Fix bug to accept analyticalStorageTTL arg

### PR DESCRIPTION
**Pull request by ilened**
**Description**
Fix `az cosmosdb sql container update` not picking up analytical-storage-ttl arg. This bug is due to code here lines 522-524 https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py . In the case when analytical-storage-ttl is the only container property specified the function will prematurely return without actually updating the value. In this scenario, users are unable to update analytical-storage-ttl using the sql cli. 

CosmosDB has recently GAed the capability for users to disable analytics by setting the analytical-storage-ttl to 0 and enable analytics for existing collections by specifying an analytical-storage-ttl. Therefore, ensuring customers are able to update the analytical-storage-ttl of their containers is important. 

**Testing Guide**
SQL cli command to update container analytical-storage-ttl property
`az cosmosdb sql container update --account-name <acc-name> --database-name <db-name> --resource-group <rg> --subscription <sub> --name <container-name> --analytical-storage-ttl <value>`

**History Notes**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

**Comment by evelyn-ys** [comment]
Can you fix CI pls? `test_cosmosdb_sql_container_update_analytical_ttl` fails.

**Comment by ilened** [comment]
> Hi Evelyn,
In the logs I see an error `knack.util.CLIError: Please run 'az login' to setup account` for the test test_cosmosdb_sql_container_update_analytical_ttl I added . I followed the testing documentation and logged in to my azure account and ran `azdev test test_cosmosdb_commands` locally to create a new recording file but I got this error posted below. Do you know what credentials or permissions I am missing? This is my first time making changes to the cli code so I am not very familiar. Thank you! 

`
azure.core.exceptions.HttpResponseError: (AuthorizationFailed) The client 'ilenedo@microsoft.com' with object id '68fd2ae3-21ec-4837-9848-7c726a3b9852' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resourcegroups/write' over scope '/subscriptions/fbe17af0-27d2-4dee-bc5e-44622527449e/resourcegroups/cli_test_cosmosdb_cassandra_keyspaceyfkvo7k2tifrtzj7hzzpgqi4vy37e76tiphhx2p' or the scope is invalid. If access was recently granted, please refresh your credentials.
E           Code: AuthorizationFailed
E           Message: The client 'ilenedo@microsoft.com' with object id '68fd2ae3-21ec-4837-9848-7c726a3b9852' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resourcegroups/write' over scope '/subscriptions/fbe17af0-27d2-4dee-bc5e-44622527449e/resourcegroups/cli_test_cosmosdb_cassandra_keyspaceyfkvo7k2tifrtzj7hzzpgqi4vy37e76tiphhx2p' or the scope is invalid. If access was recently granted, please refresh your credentials.
`

File: src/azure-cli/azure/cli/command_modules/cosmosdb/tests/latest/test_cosmosdb_commands.py [hunk 1]
```diff
@@ -614,6 +614,66 @@ def test_cosmosdb_sql_container_analytical_ttl(self, resource_group):
         container_list = self.cmd('az cosmosdb sql container list -g {rg} -a {acc} -d {db_name}').get_output_in_json()
         assert len(container_list) == 0
 
+    @ResourceGroupPreparer(name_prefix='cli_test_cosmosdb_sql_container_update_analyticalStorageTtl')
+    def test_cosmosdb_sql_container_update_analytical_ttl(self, resource_group):
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/tests/latest/test_cosmosdb_commands.py:618]: By adding new test, you also need to commit the recording yaml file

**Comment by evelyn-ys** [comment]
@ilened 
Firstly, you don't need to run all cosmosdb tests, you just need to run the exact test with `azdev test test_cosmosdb_sql_container_update_analytical_ttl`
Secondly, is sub `fbe17af0-27d2-4dee-bc5e-44622527449e` your general test sub? Please use `az account set` to switch to a sub which you are an admin or a contributor.

**Review by krishnan1985 — APPROVED** [review]

**Comment by ilened** [comment]
> > Hi Evelyn,
> > In the logs I see an error `knack.util.CLIError: Please run 'az login' to setup account` for the test test_cosmosdb_sql_container_update_analytical_ttl I added . I followed the testing documentation and logged in to my azure account and ran `azdev test test_cosmosdb_commands` locally to create a new recording file but I got this error posted below. Do you know what credentials or permissions I am missing? This is my first time making changes to the cli code so I am not very familiar. Thank you!
> 
> `azure.core.exceptions.HttpResponseError: (AuthorizationFailed) The client 'ilenedo@microsoft.com' with object id '68fd2ae3-21ec-4837-9848-7c726a3b9852' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resourcegroups/write' over scope '/subscriptions/fbe17af0-27d2-4dee-bc5e-44622527449e/resourcegroups/cli_test_cosmosdb_cassandra_keyspaceyfkvo7k2tifrtzj7hzzpgqi4vy37e76tiphhx2p' or the scope is invalid. If access was recently granted, please refresh your credentials. E Code: AuthorizationFailed E Message: The client 'ilenedo@microsoft.com' with object id '68fd2ae3-21ec-4837-9848-7c726a3b9852' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resourcegroups/write' over scope '/subscriptions/fbe17af0-27d2-4dee-bc5e-44622527449e/resourcegroups/cli_test_cosmosdb_cassandra_keyspaceyfkvo7k2tifrtzj7hzzpgqi4vy37e76tiphhx2p' or the scope is invalid. If access was recently granted, please refresh your credentials.`

**Comment by ilened** [comment]
Hi @evelyn_ys, thank you for your help! I have added the recording yaml files for the new tests and all checks have passed. Also I have some questions about how release cycles and deployments work for the azure-cli code if you don't mind. What is the date cutoff for your next release branch which this change will likely be apart of? How long does it typically take to deploy a release into prod?

**Comment by yonzhan** [comment]
You could find all milestones here: https://github.com/Azure/azure-cli/milestones

**Comment by ilened** [comment]
@evelyn-ys  @yonzhan @calvinhzy Hi can someone please review this PR and approve if satisfactory? Thank you.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**